docs/topics/ksp/ksp-additional-details.md

```diff
@@ -1,63 +1,52 @@
-structure: namely, how the reference is written. It corresponds to the
-[`type`](https://kotlinlang.org/docs/reference/grammar.html#type) element in
-Kotlin's grammar.
+A `KSTypeReference` has a `KSReferenceElement`, which models Kotlin‘s program structure: namely, how the reference is 
+written. It corresponds to the [`type`](https://kotlinlang.org/docs/reference/grammar.html#type) element in Kotlin's grammar.
 
-A `KSReferenceElement` can be a `KSClassifierReference` or
-`KSCallableReference`, which contains a lot of useful information without
-the need for resolution. For example, `KSClassifierReference` has
-`referencedName`, while `KSCallableReference` has `receiverType`,
-`functionArguments`, and `returnType`.
+A `KSReferenceElement` can be a `KSClassifierReference` or `KSCallableReference`, which contains a lot of useful 
+information without the need for resolution. For example, `KSClassifierReference` has `referencedName`, while 
+`KSCallableReference` has `receiverType`, `functionArguments`, and `returnType`.
 
-If the original declaration referenced by a `KSTypeReference` is needed,
-it can usually be found by resolving to `KSType` and accessing through
-`KSType.declaration`. Moving from where a type is mentioned to where its
-class is defined looks like this:
+If the original declaration referenced by a `KSTypeReference` is needed, it can usually be found by resolving to 
+`KSType` and accessing through `KSType.declaration`. Moving from where a type is mentioned to where its class is defined 
+looks like this:
 
 ```kotlin
-KSTypeReference -> .resolve() -> KSType -> .declaration -> KSDeclaration
+val ksType: KSType = ksTypeReference.resolve()
+val ksDeclaration: KSDeclaration = ksType.declaration
 ```
 
-Type resolution is costly and is therefore made explicit. Some of the
-information obtained from resolution is already available in
-`KSReferenceElement`. For example, `KSClassifierReference.referencedName`
-can filter out a lot of elements that are not interesting. You should
-resolve type only if you need specific information from `KSDeclaration`
-or `KSType`.
-
-Note that a `KSTypeReference` pointing to a function type has most of its
-information in its element. Although it can be resolved to the family of
-`Function0`, `Function1`, and so on, these resolutions don‘t bring any
-more information than `KSCallableReference`. One use case for resolving
-function type references is dealing with the identity of the function's
-prototype.
+Type resolution is costly and therefore has explicit form. Some of the information obtained from resolution is already 
+available in `KSReferenceElement`. For example, `KSClassifierReference.referencedName` can filter out a lot of elements 
+that are not interesting. You should resolve type only if you need specific information from `KSDeclaration` or `KSType`.
+
+Note that a `KSTypeReference` pointing to a function type has most of its information in its element. 
```

Comment:
We can use type=note, if this sentence is really a note.
Of rephrase it.

Reply:
I think "Note that a" can be just omitted here. [Commit.](https://github.com/JetBrains/kotlin-web-site/pull/2570/commits/196c8156106afc27a4dcc82c8c0d41e491ad6f53)